pallets/staking/src/lib.rs

```diff
@@ -2182,6 +2138,7 @@ impl<T: Trait> Module<T> {
         let mut all_nominators: Vec<(T::AccountId, Vec<T::AccountId>)> = Vec::new();
         let mut all_validators_and_prefs = BTreeMap::new();
         let mut all_validators = Vec::new();
+        Self::refresh_compliance_statuses();
```

Comment:
can't we simplify the process by changing the mapping to boolean based mapping and completely remove the compliance logic, It doesn't seem a required logic to me
```
pub PermissionedValidators get(permissioned_validators):
            map hasher(twox_64_concat) T::AccountId => bool;
```
and remove the `referseh_compliance_statuses()` which change the storage value and do the iteration over the permissionedvalidator list in every `select_validator()` call

& we can change the definition of `is_validator_cdd_compliant()` to 

```
pub fn is_validator_cdd_compliant(who: &T::AccountId) -> bool {
        if <PermissionedValidators<T>>::get(who) {
            Self::is_validator_or_nominator_compliant(who)
        }
        false 
    }
```

Reply:
Yep - agree it can be simplified, although I don't think it will be any more efficient. I was trying to make the minimal necessary change but I could clean up some more.

Reply:
FYI - I am going to push these now to get the functional fixes in. We can revisit the simplification above in a separate task.